src/server/middleware/metric/metric.go

```diff
@@ -1,25 +1,65 @@
 package metric
 
 import (
+	"context"
 	"net/http"
 	"strconv"
+	"strings"
 	"time"
 
 	"github.com/goharbor/harbor/src/core/config"
 	"github.com/goharbor/harbor/src/lib"
 	"github.com/goharbor/harbor/src/lib/metric"
 )
 
+// ContextOpIDKey ...
+type contextOpIDKey struct{}
+
+const (
+	// CatalogOperationID ...
+	CatalogOperationID = "v2_catalog"
+	// ListTagOperationID ...
+	ListTagOperationID = "v2_tags"
+	// ManifestOperationID ...
+	ManifestOperationID = "v2_manifest"
+	// BlobsOperationID ...
+	BlobsOperationID = "v2_blob"
+	// BlobsUploadOperationID ...
+	BlobsUploadOperationID = "v2_blob_upload"
+	// OthersOperationID ...
+	OthersOperationID = "v2_others"
+)
```

Comment:
These constants only are used in the `registry/route.go`, please move them to the `registry/route.go`.

Reply:
I was the one who suggested move them to metric pkg as they are for metrics only ...
https://github.com/goharbor/harbor/pull/14056#discussion_r562544486

But I'm fine either way.

Reply:
They are arguments of metric middleware, but at the same time they are only used for registry. 
So there are some overlap between these two categories, both of the categories are OK for me.

But when put them to registry, they shouldn't belong to route file  because they are not route. If let me choose one, I will choose `registry/util.go`